### ISSUE TITLE
Multi-target xUnit 2 to .NET Standard 2.0

### ISSUE THREAD

**Issue by teo-tsirpanis** [xunit/xunit#2553]
The xUnit 2.x packages currently target .NET Standard 1.1, which results in an enormous amount of transitive dependecies being pulled even when the underlying framework does not need them.

To reduce these dependencies, [it is recommended to also target .NET Standard 2.0 if targeting .NET Standard 1.x](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/cross-platform-targeting). Until xUnit 3 releases, it would greatly help the users of 2.x.

**Comment by omghb** [comment]
Transitive packages of `xunit.assert 2.4.1` contain vulnerabilities:

```
[net6.0-windows7.0]: 
   Transitive Package                    Resolved   Severity   Advisory URL                                     
   > System.Net.Http                     4.3.0      High       https://github.com/advisories/GHSA-7jgj-8wvc-jh57
   > System.Text.RegularExpressions      4.3.0      Moderate   https://github.com/advisories/GHSA-cmhx-cq75-c4mj
```

This could be solved by adding the `.NET Standard 2.0` as an additional target.

**Comment by maw136** [comment]
Putting my two cents, I agree all the way. Please push this upgrade. The .NET 1.x is in legacy state as well as .NET Standard 1.x. To share code with .NET Framework one can use .NET Standard 2.0 (and/or) multi-targetting.

**Comment by james-s-tayler** [comment]
For context on why the sudden flurry of activity / interest / push for supporting this Trivy, a vulnerability scanning tool, in https://github.com/aquasecurity/trivy/releases/tag/v0.30.0 implemented https://github.com/aquasecurity/trivy/pull/2487 which means NuGet packages (and their transitive dependencies) are now reporting vulnerabilities. I'm having to do a lot of upgrades for security patching at the moment and hence came across this.

**Comment by teo-tsirpanis** [comment]
xUnit used to target .NET Standard 2.0 until #1801. @bradwilson do you remember what these "`net46x` and `netstandard2.0` compatibility issues" were? Currently we target .NET Framework 4.5.2 and .NET Standard 1.1. A .NET Framework 4.6.x project would use the .NET Framework 4.5.2 assembly, with or without .NET Standard 2.0.

**Comment by james-s-tayler** [comment]
Not sure if it's related to the compatibility issues, but I do know the removal of support for `netstandard2.0` also broke a bunch of functionality that worked in `2.4.0` when you upgrade to `2.4.1` when targeting `netstandard2.0`. Related PR is here: https://github.com/xunit/xunit/pull/2419

**Comment by teo-tsirpanis** [comment]
Oh, 2.4.0 was targeting .NET Standard 2.0 and 2.4.1 stopped? _Why? 😕_

**Comment by bradwilson** [comment]
From https://docs.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-0

<img width="635" alt="image" src="https://user-images.githubusercontent.com/16944/180635718-6f3d591b-8c43-4327-af8d-5dabbd0e07c0.png">

I don't recall the details, but basically the .NET team said "oops, we thought it was compatible but there were edge cases".

**Comment by james-s-tayler** [comment]
Hmm that doesn't sound like a great reason to not target `netstandard2.0` at all though? Feels more like consumers should upgrade their projects from .NET Framework 4.6.1 to at least 4.7.2, which isn't that big of a deal even on hairy big ball of mud monolith codebases. We're consuming xunit from .NET 6 yet.

**Comment by bradwilson** [comment]
It's a HUGE undertaking to do this, because it has a lot of long-reaching impacts based on the fact that we ship multiple versions of libraries in v2 and we can't always predict which will be used.

I am unwilling to invest that effort now, since it just takes away from v3 progress. For the time being, you'll just have to live with the extra dependencies.

**Comment by teo-tsirpanis** [comment]
@bradwilson [I have created a branch](https://github.com/teo-tsirpanis/xunit/tree/netstandard2.0) that brings back .NET Standard 2.0. Can I open a PR?

It feels to me that the underlying assumptions that led to the rejection of targeting .NET Standard 2.0 are incorrect. You said in https://github.com/xunit/xunit/issues/2568#issuecomment-1233608617

> The only way to safely support .NET Standard 2.0 is to make the minimum .NET Framework version 4.7.2.

which I don't think is necessary, if we multi-target to .NET Framework alongside .NET Standard; on a .NET Framework app the SDK will pick the former binaries over the latter. [That's what the Microsoft guidelines recommend](https://learn.microsoft.com/en-us/dotnet/standard/library-guidance/cross-platform-targeting#:~:text=%E2%9C%94%EF%B8%8F%20CONSIDER%20adding%20a%20target%20for%20net461%20when%20you%27re%20offering%20a%20netstandard2.0%20target.) and in fact we already do it on most xUnit v2 packages; I did it on the rest. The content you linked on https://github.com/xunit/xunit/issues/2553#issuecomment-1193258639 does not affect us for the same reasons.

xUnit is the most popular NuGet package that only supports .NET Standard 1.x and not 2.0. Its ubiquity in the test suites of various projects has led to an explosion of transitive dependencies (the project created by `dotnet new xunit` in total depends on 92 packages which leave a footprint of 631MB, while only 15 of them weighing 86.4MB are strictly necessary) and non-actionable vulnerability false alarms.

With v3 not seeming to come any time soon and judging from the issues opened, there is a strong desire for proper .NET Standard 2.0 support in v2. I am willing to do all changes necessary to bring it in, and I am asking you to reconsider your appetite for it. Thanks.
